codec/types/interface_registry.go

```diff
@@ -101,28 +106,53 @@ type interfaceRegistry struct {
 	interfaceImpls map[reflect.Type]interfaceMap
 	implInterfaces map[reflect.Type]reflect.Type
 	typeURLMap     map[string]reflect.Type
+	signingCtx     *signing.Context
 }
 
 type interfaceMap = map[string]reflect.Type
 
 // NewInterfaceRegistry returns a new InterfaceRegistry
 func NewInterfaceRegistry() InterfaceRegistry {
-	protoFiles, err := proto.MergedRegistry()
+	registry, err := NewInterfaceRegistryWithOptions(Options{
+		ProtoFiles:            protoregistry.GlobalFiles,
+		AddressCodec:          failingAddressCodec{},
+		ValidatorAddressCodec: failingAddressCodec{},
+	})
 	if err != nil {
 		panic(err)
 	}
-	return NewInterfaceRegistryWithProtoFiles(protoFiles)
+	return registry
 }
 
-// NewInterfaceRegistryWithProtoFiles returns a new InterfaceRegistry with the specified *protoregistry.Files instance.
-func NewInterfaceRegistryWithProtoFiles(files *protoregistry.Files) InterfaceRegistry {
+type Options struct {
```

Comment:
Since this is a `types` package it seems like this should have a more specific name, perhaps `IROptions` or `InterfaceRegistryOptions` -- so that if a future type takes options, we don't have an inconsistency of one type named `Options` and other types named `BarOptions`.